sky/packages/sky/lib/src/material/scaffold.dart

```diff
@@ -39,53 +71,35 @@ class Scaffold extends StatelessComponent {
     this.body,
     this.toolBar,
     this.snackBar,
-    this.floatingActionButton
+    this.floatingActionButton,
+    this.bottomSheet
   }) : super(key: key);
 
   final Widget body;
   final ToolBar toolBar;
   final Widget snackBar;
   final Widget floatingActionButton;
+  final Widget bottomSheet;
 
   Widget build(BuildContext context) {
-    final ToolBar paddedToolBar = toolBar?.withPadding(new EdgeDims.only(top: ui.window.padding.top));
+    final Widget paddedToolBar = toolBar?.withPadding(new EdgeDims.only(top: ui.window.padding.top));
     final Widget materialBody = body != null ? new Material(child: body) : null;
-    Widget toolBarAndBody;
-    if (paddedToolBar != null && materialBody != null)
-      toolBarAndBody = new CustomMultiChildLayout(<Widget>[materialBody, paddedToolBar],
-        delegate: _toolBarAndBodyLayout
-      );
-    else
-      toolBarAndBody = paddedToolBar ?? materialBody;
-
-    final List<Widget> bottomColumnChildren = <Widget>[];
-
-    if (floatingActionButton != null)
-      bottomColumnChildren.add(new Padding(
-        // TODO(eseidel): These change based on device size!
-        padding: const EdgeDims.only(right: 16.0, bottom: 16.0),
-        child: floatingActionButton
-      ));
-
-    // TODO(jackson): On tablet/desktop, minWidth = 288, maxWidth = 568
+    Widget constrainedSnackBar;
     if (snackBar != null) {
-      bottomColumnChildren.add(new ConstrainedBox(
+      // TODO(jackson): On tablet/desktop, minWidth = 288, maxWidth = 568
+      constrainedSnackBar = new ConstrainedBox(
         constraints: const BoxConstraints(maxHeight: kSnackBarHeight),
         child: snackBar
-      ));
-    }
-
-    final List<Widget> stackChildren = <Widget>[toolBarAndBody];
-
-    if (bottomColumnChildren.length > 0) {
-      stackChildren.add(new Positioned(
-        right: 0.0,
-        left: 0.0,
-        bottom: 0.0,
-        child: new Column(bottomColumnChildren, alignItems: FlexAlignItems.end)
-      ));
+      );
     }
-
-    return new Stack(stackChildren);
+    return new CustomMultiChildLayout(<Widget>[
+      materialBody ?? new Container(height: 0.0),
+      paddedToolBar ?? new Container(height: 0.0),
+      bottomSheet ?? new Container(height: 0.0),
+      constrainedSnackBar ?? new Container(height: 0.0),
+      floatingActionButton ?? new Container(height: 0.0)
```

Comment:
I agree. I will add some support for identifying children by name and discovering which ones exist in a separate CL